### PR TITLE
Registrar - name widget

### PR DESCRIPTION
**Pull request by HejdaJakub**
* There was a problem, that we were not able to parse names and titles from displayName always correctly.
* I created a back-end part of a new feature called name widget, that allows user to put names and titles to application manually, but correctly.

### CODE REVIEW

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java [hunk 1]
```diff
@@ -1702,6 +1703,25 @@ public Application approveApplicationInternal(PerunSession sess, int appId) thro
...
+						User user = usersManager.getUserById(registrarSession, app.getUser().getId());
+						if (item.getValue() != null && !item.getValue().isEmpty()) {
+							// user's attributes in JSON format from name widget
+							JSONObject fullName = new JSONObject(item.getValue());
+							user.setTitleBefore(fullName.getString("titleBefore"));
+							user.setFirstName(fullName.getString("firstName"));
+							user.setMiddleName(fullName.getString("middleName"));
+							user.setLastName(fullName.getString("lastName"));
+							user.setTitleAfter(fullName.getString("titleAfter"));
+							usersManager.updateUser(registrarSession, user);
```
Comment (Vojtech-Sassmann) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java:1720]: You should also set the updated user to the application object.

**Review by Vojtech-Sassmann — CHANGES_REQUESTED** [review]

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java [hunk 2]
```diff
@@ -1702,6 +1703,26 @@ public Application approveApplicationInternal(PerunSession sess, int appId) thro
 					}
 				}
 
+				// get all app items
+				List<ApplicationFormItemData> items = getApplicationDataById(registrarSession, app.getId());
```
Comment (balcirakpeter) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java:1707]: I do not think that the place where this code is used is right. You want to set names only when you are creating new users.
This way you could re-set them when some user is applying to another Vo.

**Review by balcirakpeter — CHANGES_REQUESTED** [review]

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java (hunk 2 above)
Comment (Vojtech-Sassmann) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java:1707]: Actually, I have discussed this with @licehammer, and for this widget, we want it to set the names even to existing users.

**Comment by HejdaJakub** [comment]
We have decided to deal with this problem only in GUI, so back-end changes are not needed.

**Pull request closed without merging**